aten/src/ATen/MapAllocator.cpp

```diff
@@ -292,6 +292,28 @@ MapAllocator::MapAllocator(WithFd, std::string_view filename, int fd, int flags,
           if (ftruncate(fd, static_cast<off_t>(size)) == -1) {
             TORCH_CHECK(false, "unable to resize file <", filename_, "> to the right size: ", c10::utils::str_error(errno), " (", errno, ")");
           }
+
+#if !defined(__APPLE__) && !defined(__ANDROID__)
```

Comment:
Would it be a better to use a CMake check function exists? Really no better way to check if it exists? Or no more portable way to preallocate it?